File: rosettacode_org/S/Sorting_algorithms/Insertion_sort/Insertion_sort.py

def insertion_sort(L):
    for i in range(1, len(L)):
        j = i - 1
        key = L[i]
        key = L[i]
        while (L[j] > key) and (j >= 0):
            L[j+1] = L[j]
            j -= 1
        L[j+1] = key

L = [4, 65, 2, -31, 0, 99, 83, 782, 1]
insertion_sort(L)
print(L)

File: algorithms/ondrej-kvasnovsky-2/algorithms/src/main/java/org/example/sorting/InsertionSort.java
package org.example.sorting;

import java.util.Arrays;

public class InsertionSort {
    public static void insertionSort(int[] array) {
        int j;
        for(int i = 1; i < array.length; i++) {
            j = i;
            while((j > 0) && (array[j] < array[j - 1])) {
                swap(array, j, j - 1);
                j = j - 1;
                System.out.println(Arrays.toString(array));
            }
        }
    }

    public static void swap(int[] array, int i, int j) {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }

    public static void main(String[] args) {
        int[] array = {5, 4, 3, 2, 1};
        insertionSort(array);
    }
}
